server/authzhandler.go

```diff
@@ -33,6 +35,13 @@ type Authzhandler struct {
 }
 
 func (s *Authzhandler) ServeHTTP(w http.ResponseWriter, req *http.Request) {
+	body, err := ioutil.ReadAll(req.Body)
```

Comment:
is req.Body being closed?

https://pkg.go.dev/net/http#pkg-overview

The client must close the response body when finished with it:
```go
resp, err := http.Get("http://example.com/")
if err != nil {
	// handle error
}
defer resp.Body.Close()
body, err := io.ReadAll(resp.Body)
// ...
```